### PR TITLE
WT-6263 Reenable history store verification

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/btree/bt_vrfy.c [hunk 1]
```diff
@@ -170,30 +170,24 @@ __wt_verify(WT_SESSION_IMPL *session, const char *cfg[])
     const char *name;
     bool bm_start, quit;
 
-#if 0
-    /* FIXME-WT-6263: Temporarily disable history store verification. */
     uint32_t session_flags;
```
Comment (tetsuo-cpp) [inline src/btree/bt_vrfy.c:173]: Can we move these back into the declaration block above?
The only reason I moved them down here is so I could `#ifdef` them out as a group.

File: test/format/util.c [hunk 2]
```diff
@@ -165,15 +165,13 @@ fclose_and_clear(FILE **fpp)
 void
 timestamp_once(WT_SESSION *session)
 {
-    static const char *oldest_timestamp_str = "oldest_timestamp=";
     WT_CONNECTION *conn;
     WT_DECL_RET;
-    char buf[WT_TS_HEX_STRING_SIZE + 64];
+    char ts_hex_buf[WT_TS_HEX_STRING_SIZE];
+    char tscfg[64];
```
Comment (tetsuo-cpp) [inline test/format/util.c:171]: Is this still enough? `WT_TS_HEX_STRING_SIZE` seems big.
I think if we're following the previous convention, it should be `64 + 2 * WT_TS_HEX_STRING_SIZE`.

Comment (quchenhao) [reply test/format/util.c:171]: 64 bytes should be enough, we use the same value in timestamp abort and other tests. In total I calculated that we need 55 bytes

File: src/btree/bt_vrfy.c (hunk 1 above)
Comment (quchenhao) [reply src/btree/bt_vrfy.c]: fixed

**Comment by quchenhao** [comment]
@agorrod @bvpvamsikrishna 
With the change, we get

```
[2020/06/09 02:08:51.699] ############################################
 [2020/06/09 02:08:51.699] #  RUN PARAMETERS: V2
 [2020/06/09 02:08:51.699] ############################################
 [2020/06/09 02:08:51.699] assert.commit_timestamp=0
 [2020/06/09 02:08:51.699] assert.read_timestamp=0
 [2020/06/09 02:08:51.699] backup=0
 [2020/06/09 02:08:51.699] backup.incremental=off
 [2020/06/09 02:08:51.699] backup.incr_granularity=12833
 [2020/06/09 02:08:51.699] btree.bitcnt=3
 [2020/06/09 02:08:51.699] btree.compression=none
 [2020/06/09 02:08:51.699] btree.dictionary=0
 [2020/06/09 02:08:51.699] btree.huffman_key=0
 [2020/06/09 02:08:51.699] btree.huffman_value=1
 [2020/06/09 02:08:51.699] btree.internal_key_truncation=1
 [2020/06/09 02:08:51.699] btree.internal_page_max=16
 [2020/06/09 02:08:51.699] btree.key_gap=10
 [2020/06/09 02:08:51.699] btree.key_max=127
 [2020/06/09 02:08:51.699] btree.key_min=25
 [2020/06/09 02:08:51.699] btree.leaf_page_max=17
 [2020/06/09 02:08:51.699] btree.memory_page_max=10
 [2020/06/09 02:08:51.699] btree.prefix_compression=1
 [2020/06/09 02:08:51.699] btree.prefix_compression_min=0
 [2020/06/09 02:08:51.699] btree.repeat_data_pct=77
 [2020/06/09 02:08:51.699] btree.reverse=0
 [2020/06/09 02:08:51.699] btree.split_pct=73
 [2020/06/09 02:08:51.699] btree.value_max=3038
 [2020/06/09 02:08:51.699] btree.value_min=11
 [2020/06/09 02:08:51.699] cache=100
 [2020/06/09 02:08:51.699] cache.evict_max=4
 [2020/06/09 02:08:51.699] cache.minimum=0
 [2020/06/09 02:08:51.699] checkpoint=on
 [2020/06/09 02:08:51.699] checkpoint.log_size=147
 [2020/06/09 02:08:51.699] checkpoint.wait=84
 [2020/06/09 02:08:51.699] disk.checksum=uncompressed
 [2020/06/09 02:08:51.699] disk.data_extend=0
 [2020/06/09 02:08:51.699] disk.direct_io=0
 [2020/06/09 02:08:51.699] disk.encryption=none
 [2020/06/09 02:08:51.699] disk.firstfit=0
 [2020/06/09 02:08:51.699] disk.mmap=1
 [2020/06/09 02:08:51.699] disk.mmap_all=0
 [2020/06/09 02:08:51.699] format.abort=0
 [2020/06/09 02:08:51.699] format.independent_thread_rng=0
 [2020/06/09 02:08:51.699] format.major_timeout=0
 [2020/06/09 02:08:51.699] logging=0
 [2020/06/09 02:08:51.699] logging.archive=1
 [2020/06/09 02:08:51.699] logging.compression=none
 [2020/06/09 02:08:51.699] logging.file_max=121983
 [2020/06/09 02:08:51.699] logging.prealloc=1
 [2020/06/09 02:08:51.699] lsm.auto_throttle=1
 [2020/06/09 02:08:51.699] lsm.bloom=1
 [2020/06/09 02:08:51.699] lsm.bloom_bit_count=58
 [2020/06/09 02:08:51.699] lsm.bloom_hash_count=10
 [2020/06/09 02:08:51.699] lsm.bloom_oldest=0
 [2020/06/09 02:08:51.699] lsm.chunk_size=3
 [2020/06/09 02:08:51.699] lsm.merge_max=13
 [2020/06/09 02:08:51.699] lsm.worker_threads=4
 [2020/06/09 02:08:51.699] ops.alter=0
 [2020/06/09 02:08:51.699] ops.compaction=0
 [2020/06/09 02:08:51.699] ops.hs_cursor=1
 [2020/06/09 02:08:51.700] ops.pct.delete=16
 [2020/06/09 02:08:51.700] ops.pct.insert=51
 [2020/06/09 02:08:51.700] ops.pct.modify=0
 [2020/06/09 02:08:51.700] ops.pct.read=12
 [2020/06/09 02:08:51.700] ops.pct.write=21
 [2020/06/09 02:08:51.700] ops.prepare=0
 [2020/06/09 02:08:51.700] ops.random_cursor=0
 [2020/06/09 02:08:51.700] ops.rebalance=1
 [2020/06/09 02:08:51.700] ops.salvage=1
 [2020/06/09 02:08:51.700] ops.truncate=1
 [2020/06/09 02:08:51.700] ops.verify=1
 [2020/06/09 02:08:51.700] quiet=1
 [2020/06/09 02:08:51.700] runs=1
 [2020/06/09 02:08:51.700] runs.in_memory=0
 [2020/06/09 02:08:51.700] runs.ops=50000
 [2020/06/09 02:08:51.700] runs.rows=10000
 [2020/06/09 02:08:51.700] runs.source=table
 [2020/06/09 02:08:51.700] runs.threads=4
 [2020/06/09 02:08:51.700] runs.timer=360
 [2020/06/09 02:08:51.700] runs.type=row-store
 [2020/06/09 02:08:51.700] runs.verify_failure_dump=0
 [2020/06/09 02:08:51.700] statistics=0
 [2020/06/09 02:08:51.700] statistics.server=0
 [2020/06/09 02:08:51.700] stress.aggressive_sweep=0
 [2020/06/09 02:08:51.700] stress.checkpoint=0
 [2020/06/09 02:08:51.700] stress.hs_checkpoint_delay=0
 [2020/06/09 02:08:51.700] stress.hs_sweep=0
 [2020/06/09 02:08:51.700] stress.split_1=0
 [2020/06/09 02:08:51.700] stress.split_2=0
 [2020/06/09 02:08:51.700] stress.split_3=0
 [2020/06/09 02:08:51.700] stress.split_4=0
 [2020/06/09 02:08:51.700] stress.split_5=0
 [2020/06/09 02:08:51.700] stress.split_6=0
 [2020/06/09 02:08:51.700] stress.split_7=0
 [2020/06/09 02:08:51.700] stress.split_8=0
 [2020/06/09 02:08:51.700] transaction.frequency=100
 [2020/06/09 02:08:51.700] transaction.isolation=snapshot
 [2020/06/09 02:08:51.700] transaction.timestamps=1
 [2020/06/09 02:08:51.700] wiredtiger.config=
 [2020/06/09 02:08:51.700] wiredtiger.rwlock=1
 [2020/06/09 02:08:51.700] wiredtiger.leak_memory=0
 [2020/06/09 02:08:51.700] ############################################
 [2020/06/09 02:08:51.700] t: FAILED: wts_rebalance/55: session->rebalance(session, g.uri, NULL): Device or resource busy
```
I'm not sure how hs verification interferes rebalance and I don't know much about rebalance.

File: test/csuite/schema_abort/main.c [hunk 3]
```diff
@@ -478,7 +478,7 @@ thread_ts_run(void *arg)
              * availability at older timestamps.
              */
```
Comment (quchenhao) [inline test/csuite/schema_abort/main.c:479]: @tetsuo-cpp These tests they all set the buffer size to 64.

File: test/csuite/schema_abort/main.c [hunk 4]
```diff
@@ -478,7 +478,7 @@ thread_ts_run(void *arg)
...
             testutil_check(__wt_snprintf(tscfg, sizeof(tscfg),
-              "oldest_timestamp=%" PRIx64 ",stable_timestamp=%" PRIx64, oldest_ts, oldest_ts));
```
Comment (agorrod) [inline test/csuite/schema_abort/main.c:481]: Why did you change the order of these? We generally alphabeticalize things where possible, and I think that makes sense here too.

Comment (quchenhao) [reply test/csuite/schema_abort/main.c:481]: OK, that just feel counterintuitive to me. I think we should move stable timestamp first and then oldest timestamp. If the sequence doesn't matter, I'll respect the alphabetic order.

**Comment by keithbostic** [comment]
@quchenhao, I've rebased your `WT-6263 Reenable history store verification` PR against the `rts-test-format` branch, and resolved the conflicts.

(The only conflict was your change to configure setting the stable timestamp in the format test program. As the `rts-test-format` branch already updates the stable timestamp based on an earlier merge from @ddanderson, I removed your changes.)

I've unblocked the WT-6263 ticket as well.

@bvpvamsikrishna, @agorrod, @tetsuo-cpp: you're all queued up for review of this ticket now that it's unblocked.

**Comment by quchenhao** [comment]
@agorrod @bvpvamsikrishna Can we get this merged? It would be good if we can reenable hs verification again.
